packages/astro/src/core/render/params-and-props.ts

```diff
@@ -64,7 +64,7 @@ function getRouteParams(route: RouteData, pathname: string): Params | undefined
 	if (route.params.length) {
 		// The RegExp pattern expects a decoded string, but the pathname is encoded
 		// when the URL contains non-English characters.
-		const paramsMatch = route.pattern.exec(decodeURIComponent(pathname));
+		const paramsMatch = route.pattern.exec(decodeURI(pathname));
```

Comment:
We've had many URI encoding issues in the past, so we tend to be very cautious when modifying any logic related to this.
 
Seems like using `decodeURI` here would break users who have encoded `, ? : @ & = + $ #` but potentially fix just users who have encoded `/`?

Reply:
Yes, the change the behavior for users who have those symbols encoded in their parameter. But, it prevents encoded components from being interrupted as query params, uri params, location hash, etc... My gut is the change here is the right long-term strategy to prevent that class of bugs. (I also think may be issues with cache misses for these cases.)

I did think about adding automatic uri component decoding when accessing the parameter itself. However, I couldn't find a great spot to do that. The params object are used as part of the URI cache today, so, I can't encode it there and still have the cache work. There might be a spot to add this when accessing the value, but I don't know if that would cause other inconsistencies.

The behavior around this one was changed in a [minor release](https://github.com/withastro/astro/commit/4c7ba4da084d7508df91cbac03c2b099a8301e2b) in the past (i.e. from not decoded to decoded), so I leaned towards not making a more major refactor to keep backwards compatibility.

What do you think?

Reply:
> Yes, the change the behavior for users who have those symbols encoded in their parameter

This will cause a breaking change for users who rely on `, ? : @ & = + $ #`. Not sure we can land this PR as is 🤔